drivers/pwm/pwm_ite_it8801.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2024 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8801_pwm
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/mfd/mfd_ite_it8801.h>
+#include <zephyr/drivers/pwm.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(pwm_ite_it8801, CONFIG_PWM_LOG_LEVEL);
+
+struct it8801_pwm_map {
+	int pushpull_en;
+};
+const static struct it8801_pwm_map it8801_pwm_gpio_map[] = {
+	[1] = {.pushpull_en = BIT(0)}, [2] = {.pushpull_en = BIT(1)}, [3] = {.pushpull_en = BIT(2)},
+	[4] = {.pushpull_en = BIT(3)}, [7] = {.pushpull_en = BIT(4)}, [8] = {.pushpull_en = BIT(5)},
+	[9] = {.pushpull_en = BIT(6)},
+};
+
+struct it8801_mfd_pwm_altctrl_cfg {
+	/* GPIO control device structure */
+	const struct device *gpiocr;
+	/* GPIO control pin */
+	uint8_t pin;
+	/* GPIO function select */
+	uint8_t alt_func;
+};
+
+struct pwm_it8801_config {
+	/* IT8801 controller dev */
+	const struct device *mfd;
+	/* PWM alternate configuration */
+	const struct it8801_mfd_pwm_altctrl_cfg *altctrl;
+	int mfdctrl_len;
+	int channel;
+	/* PWM mode control register */
+	uint8_t reg_mcr;
+	/* PWM duty cycle register */
+	uint8_t reg_dcr;
+	/* PWM prescale LSB register */
+	uint8_t reg_prslr;
+	/* PWM prescale MSB register */
+	uint8_t reg_prsmr;
+};
+
+struct pwm_it8801_data {
+	/* I2C device for the MFD parent */
+	const struct i2c_dt_spec *i2c_dev;
```

Comment:
`ite,it8801-pwm` is a child node of the mfd device right? Then can you get this with `DEVICE_DT_GET(DT_PARENT(...))` at build time?

Reply:
That's right, thanks for the tip, we can use this: I2C_DT_SPEC_GET(DT_INST_PARENT(inst))